### DISCUSSION TITLE
Save a Node Template with a Group

### DISCUSSION THREAD

**Discussion by JorgeR81** [comfyanonymous/ComfyUI#2936] [Q&A] [answered]
Is there a way to save and load a node template, with a group included ?

Thanks

**Comment by kakachiex2** [comment]
Men I want this so bad, another good feature will be to move a master group and all other container group inside; wen you have a group inside another it only move the selected group and not the other inside.

**Reply by aomiks** [reply]
组目前也只能一个一个的移动，不能像节点可以一起移动

**Comment by kakachiex2** [comment]
In this workflow I organized everything by groups look what happen if I move the master group

https://github.com/comfyanonymous/ComfyUI/assets/1762112/fbe021be-7749-409c-8387-6e377fdf1200

**Comment by JorgeR81** [comment]
> another good feature will be to move a master group and all other container group inside

Yes, that would be nice,
Especially because you can't just move the small groups back into place. 
The groups start to push the nodes as soon as they make contact ...

But on a side note, 
I see that you're using the Fast Groups Bypasser (rgthree) with cg-use-everywhere.
Is that working for you ? 
When I use the rgthree node to bypass a group, the UE wires don't seem to update.

**Reply by kakachiex2** [reply]
Yes is working

**Reply by kakachiex2** [reply]
It stop working wen you have two use-everywhere node with the same node for example two vae loader it stop working

**Reply by JorgeR81** [reply]
OK, thanks.
I tried this again and it worked !

The problem is that the UE visualizations are not refreshed, when an rgtree node is used, to bypass/mute a node/group.

<details>
<summary>show image ( groups )</summary>
<IMG src="https://github.com/comfyanonymous/ComfyUI/assets/72734038/f3bd2d92-7aff-48e0-97c9-d61aadcbc918"  alt="image.jpg"/>
</details>

<details>
<summary>show image ( nodes )</summary>
<IMG src="https://github.com/comfyanonymous/ComfyUI/assets/72734038/57890686-69a5-40b3-827c-916792f85723"  alt="image.jpg"/>
</details>

I need to move or left click on a node, for this to update. 
But when I bypass or mute via right click, it shows immediately.

I think I will open an issue for rgthree or cg-use-everywhere, to see if this can be polished. 

These template / group improvements, plus with these customs nodes, it would be a powerful combination !

**Reply by kakachiex2** [reply]
your right

**Comment by JorgeR81** [comment]
I've just found that there is an issue and a pull request to add this feature. 

I guess I will keep this open, to help people find these, via the discussion search.

Or if someone knows about a workaround with custom nodes, until this it's implemented in Comfy UI

https://github.com/comfyanonymous/ComfyUI/issues/1912 

https://github.com/comfyanonymous/ComfyUI/pull/2774

**Reply by kakachiex2** [reply]
nice

**Comment by JorgeR81** [comment, marked as answer]
I've just found a workaround with [ **_aegisflow_utility_nodes_** ].
It creates a new group when you load a template. 
https://github.com/aegis72/aegisflow_utility_nodes?tab=readme-ov-file#auto-grouping-of-nodes-placed-as-templates

It also has other functions that complement [ **_cg-use-everywhere_** ].
The "pass" nodes allow me to bypass nodes without breaking the UE links.
https://github.com/chrisgoringe/cg-use-everywhere?tab=readme-ov-file#cautions

![pass](https://github.com/comfyanonymous/ComfyUI/assets/72734038/afbeef95-7bb4-4c37-b52c-645413045478)
